unstructured_inference/inference/layout.py

```diff
@@ -89,9 +91,20 @@ def from_file(
         logger.info(f"Reading PDF for file: {filename} ...")
         layouts, images = load_pdf(filename, load_images=True)
         if len(layouts) > len(images):
-            raise RuntimeError(
-                "Some images were not loaded. Check that poppler is installed and in your $PATH."
-            )
+            try:
+                images = []
+                for page in fitz.open(filename):
+                    pixmap = page.get_pixmap()
+                    images.append(Image.open(io.BytesIO(pixmap.pil_tobytes("ppm"))))
+                if len(layouts) > len(images):
+                    raise ValueError("PyMuPDF extracted less images than layouts.")
+
+            except Exception as exc:
+                raise RuntimeError(
+                    "Some images were not loaded. "
+                    "Check that poppler is installed and in your $PATH."
+                ) from exc
```

Comment:
Hi @cragwolfe I took a look at the suggestion. I think an import-level warning may not actually do the job we desire here since we are actually raising an error here. Hence, the extraction process is expected to end regardless of whether we have a warning or not.